cms/grading/languages/php.py

```diff
@@ -40,11 +40,19 @@ def source_extensions(self):
         """See Language.source_extensions."""
         return [".php"]
 
+    @property
+    def executable_extension(self):
+        """See Language.executable_extension."""
+        return ".php"
+
     def get_compilation_commands(self,
                                  source_filenames, executable_filename,
                                  for_evaluation=True):
         """See Language.get_compilation_commands."""
-        return [["/bin/cp", source_filenames[0], executable_filename]]
+        if source_filenames[0] != executable_filename:
+            return [["/bin/cp", source_filenames[0], executable_filename]]
+        else:
+            return [["/bin/true"]]
```

Comment:
Can this be an empty list instead?

Reply:
There is a problem here: https://github.com/cms-dev/cms/blob/d4c9e926bd52d8022069c417b206b0882ef4d1ba/cms/grading/steps/utils.py#L88-L89

If there are no commands, then `stats` is `None`. I can see two possible ways: initialize `stats` with a "zero" object (or treat `None` as zero in relevant places), or skip compilation altogether if there are no commands. But the latter is likely to introduce special cases in multiple places.

This PR avoids the problem using `/bin/true` for now.

Reply:
Ok, please add a comment saying why we need it.

Reply:
Added a comment.